src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1551,11 +1554,11 @@ impl Disambiguator {
                 "value" => NS(Namespace::ValueNS),
                 "macro" => NS(Namespace::MacroNS),
                 "prim" | "primitive" => Primitive,
-                _ => return find_suffix(),
+                _ => return Err(format!("unknown disambiguator `{}`", prefix)),
```

Comment:
Why was the previous behavior of this to look at the suffix if there the prefix was not recognized? I ran the testsuite and everything passed with this change.

Reply:
No particular reason - that was the behavior before I refactored it and I just kept it the same. I think it shouldn't ever change the behavior since `@` is invalid in paths.